generate/genfile/genfile.go

```diff
@@ -130,6 +132,11 @@ func Load(projmeta project.Metadata, sm stack.Metadata, globals globals.G) ([]Fi
 			Str("origin", origin).
 			Logger()
 
+		err := lets.Load(genFileBlock.lets, evalctx.Context)
```

Comment:
Inside this loop we call Load with the same evalctx.Context multiple times for each genFileBlock without cleaning up the evalctx.Context "lets" namespace, at least I didn't find where we do this. If we don't cleanup won't the lets namespace loaded from a generate_file block be available to another generate_file block ?

lets.Load only sets the namespace to an empty map if the namespace doesn't exist, so it seems that when it already exists it will only add more (or override) what is already there from previous loads. Sorry if I missed where it is cleanup up, if that is the case just disregard the comment.

Reply:
On generate_hcl we create a new context per generate_hcl block, maybe we could do the same here (and also have some tests to cover this, the test exists on generate_hcl but I think it is missing here).

Reply:
yep I fixed this in genhcl but forgot here. Thanks.